qt-models/filterconstraintmodel.cpp

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "filterconstraintmodel.h"
+#include "core/qthelper.h"	// for timestamp conversions
+#include <QTime>
+
+FilterConstraintModel::~FilterConstraintModel()
+{
+}
+
+// QTime <-> seconds in integer conversion functions
+static QTime secondsToTime(int seconds)
+{
+	return QTime::fromMSecsSinceStartOfDay(seconds * 1000);
+}
+
+static int timeToSeconds(const QTime &t)
+{
+	return t.msecsSinceStartOfDay() / 1000;
+}
+
+QVariant FilterConstraintModel::data(const QModelIndex &index, int role) const
+{
```

Comment:
are we going to avoid being bitten by columns / role because we do this in QML everywhere?

Reply:
The problem is avoided by not using tables and there therefore are no columns. The QWidget pseudo-UI, which I implemented, can access the model just fine. Of course, if you would make *only* a QWidget UI, then you wouldn't use this model thing in the first place, but interact with C++ classes directly. *sigh*